Inherit and Override Page Context

DeveloperMetal opened the pull request:
Hi Frappe Team!

This is a small contribution. A little while back I was trying to override existing page contexts on erpnext and found that only template files were allowed to be overwritten.

I've implemented a simple hook which allow overriding existing page controllers while inheriting the work the controller does:

ex:

extend_controller_context = [ "erpnext.templates.pages.cart": "mymodule.templates.pages.cart" ] 

The above lets "erpnext.template.pages.cart" run and passes the context to "mymodule.templates.pages.cart" to extend the cart page with extra content for the template.

DeveloperMetal committed:
Added inheritable context hook

DeveloperMetal committed:
Merge branch 'develop' into inherit_context

rmehta commented on frappe/website/context.py:
```diff
@@ -62,22 +81,18 @@ def build_context(context):
+		
+		# controller context extensions
+		context_controller_hooks = frappe.get_hooks("extend_controller_context") or {}
+		for controller, extension in context_controller_hooks.items():
```
can we call this `extend_website_page_controller_context`, because controller by default means doctype controller

icloakbuilder replied:
Yeah that sounds better.
I've updated the commit with the hook name change.

Commit:
Renamed hook

Renamed hook to extend_website_page_controller_context to better fit the definition of a website page controller.
